pkgs/applications/misc/blender/default.nix

```diff
@@ -1,142 +1,130 @@
-{ config, stdenv, lib, fetchurl, fetchzip, boost, cmake, ffmpeg, gettext, glew
-, ilmbase, libXi, libX11, libXext, libXrender
-, libjpeg, libpng, libsamplerate, libsndfile
-, libtiff, libGLU, libGL, openal, opencolorio, openexr, openimagedenoise, openimageio2, openjpeg, python39Packages
-, openvdb, libXxf86vm, tbb, alembic
-, zlib, fftw, opensubdiv, freetype, jemalloc, ocl-icd, addOpenGLRunpath
-, jackaudioSupport ? false, libjack2
-, cudaSupport ? config.cudaSupport or false, cudatoolkit
-, colladaSupport ? true, opencollada
-, spaceNavSupport ? stdenv.isLinux, libspnav
-, makeWrapper
-, pugixml, llvmPackages, SDL, Cocoa, CoreGraphics, ForceFeedback, OpenAL, OpenGL
-, potrace
-, openxr-loader
-, embree, gmp, libharu
+{ stdenv, lib, config, fetchzip, fetchurl, cmake, makeWrapper
+# Common:
+, alembic, boost, embree, ffmpeg, fftw, freetype, gettext
+, glew, gmp, ilmbase, jemalloc, libharu, libjpeg, libpng
+, libsamplerate, libsndfile, libtiff, opencolorio, openexr
+, openimagedenoise, openimageio2, openjpeg, opensubdiv
+, potrace, pugixml, python3, tbb, zlib, zstd
+
+# Linux specific:
+, libGL, libGLU, libX11, libXext, libXi, libXrender, libXxf86vm
+, openxr-loader, openvdb, openal
+
+# Darwin specific:
+, Cocoa, CoreGraphics, ForceFeedback, llvmPackages, OpenAL, OpenGL, SDL
+
+# Optional:
+, jackaudioSupport ? false
+, libjack2
+
+, cudaSupport ? config.cudaSupport or false
+, cudatoolkit_11, addOpenGLRunpath
+
+, colladaSupport ? true
+, opencollada
+
+, spaceNavSupport ? stdenv.isLinux
+, libspnav
+
+, waylandSupport ? false
+, dbus, libffi, libxkbcommon, pkg-config, wayland, wayland-protocols
 }:
 
-with lib;
 let
-  python = python39Packages.python;
-  optix = fetchzip {
-    url = "https://developer.download.nvidia.com/redist/optix/v7.0/OptiX-7.0.0-include.zip";
-    sha256 = "1b3ccd3197anya2bj3psxdrvrpfgiwva5zfv2xmyrl73nb2dvfr7";
-  };
-
+  numpyPath = "${python3.pkgs.numpy}/${python3.sitePackages}";
 in
 stdenv.mkDerivation rec {
   pname = "blender";
-  version = "2.93.5";
+  version = "3.0.0";
 
+  # We use fetchurl here instead of fetchzip,
+  # so that source is available at the https://tarballs.nixos.org
   src = fetchurl {
     url = "https://download.blender.org/source/${pname}-${version}.tar.xz";
     sha256 = "1fsw8w80h8k5w4zmy659bjlzqyn5i198hi1kbpzfrdn8psxg2bfj";
```

Comment:
that's not the right hash any more

Reply:
Ah, glad I am not the only one.

Reply:
Sorry, my fault, i've messed up when rebasing, ugh :1st_place_medal: